cpp/src/arrow/array-test.cc

```diff
@@ -456,12 +484,32 @@ TYPED_TEST(TestPrimitiveBuilder, TestAppendNull) {
     ASSERT_OK(this->builder_->AppendNull());
   }
 
-  std::shared_ptr<Array> result;
-  ASSERT_OK(this->builder_->Finish(&result));
+  std::shared_ptr<Array> out;
+  ASSERT_OK(this->builder_->Finish(&out));
+  auto result = std::dynamic_pointer_cast<typename TypeParam::ArrayType>(out);
```

Comment:
Why is the cast necessary?

Reply:
Left-over from the older implementation, I've removed it.